src/BSON/Timestamp.c

```diff
@@ -54,7 +54,7 @@ zend_object_handlers php_phongo_handler_timestamp;
 
 /* Initialize the object and return whether it was successful. An exception will
  * be thrown on error. */
-static bool php_phongo_timestamp_init(php_phongo_timestamp_t *intern, phongo_long increment, phongo_long timestamp TSRMLS_DC)
+static bool php_phongo_timestamp_init(php_phongo_timestamp_t *intern, int64_t increment, int64_t timestamp TSRMLS_DC)
```

Comment:
I assume you mean the `< 0` comparisons would fail?

Since `php_phongo_timestamp_init_from_string()` was always parsing the strings into `int64_t` values, I suppose there was possible degredation there. The other code path is `php_phongo_timestamp_init_from_hash()` picking up `IS_LONG` values. In that case, would a negative `int32_t` value expand to a negative `int64_t` value (essentially the sign bit relocates) and still yield an exception?